upload/src/addons/SV/NoticeTimeReplacable/_files/js/sv/notice-time-replacable/core.js

```diff
@@ -0,0 +1,131 @@
+var SV = window.SV || {};
+SV.NoticeTimeReplacable = SV.NoticeTimeReplacable || {};
+
+!function($, window, document, _undefined)
+{
+    "use strict";
+
+    SV.NoticeTimeReplacable.RelativeTimestamp = XF.Element.newHandler({
+        options: {
+            timestamp: null,
+        },
+
+        timer: null,
+
+        init: function()
+        {
+            if (!this.options.timestamp)
+            {
+                console.error('Timestamp is missing.');
+                return;
+            }
+
+            this.timer = setInterval(XF.proxy(this, 'updateTime'), 1000);
+        },
+
+        /**
+         * @param {Number|String} value
+         * @param {String} phrase
+         */
+        getDatePart: function (value, phrase)
+        {
+            if (typeof value === 'string')
+            {
+                value = parseInt(value) || 0;
+            }
+
+            if (typeof phrase !== 'string' || !phrase)
+            {
+                console.error('Invalid phrase provided.');
+                clearInterval(this.timer);
+                return false;
+            }
+
+            var translatedValue = XF.phrase('svNoticeTimeReplacables_' + phrase + (value > 1 ? 's' : ''), {
+                '{count}': value
+            }, null);
+
+            if (translatedValue === null)
+            {
+                console.error('Invalid phrase provided.');
+                return false;
+            }
+
+            return translatedValue;
+        },
+
+        updateTime: function ()
+        {
+            var now = Math.floor(Date.now() / 1000) * 1000,
+                end = this.options.timestamp * 1000,
+                momentObj,
+                timeArr = [];
+
+            if (now <= end)
+            {
+                momentObj = moment.duration(end - now, 'milliseconds');
+            }
+            else
+            {
+                var $noticeContent = this.$target.closest('.notice-content'),
+                    $noticeDismissButton = $noticeContent.length() ? $noticeContent.find('.notice-dismiss') : null;
+
+                if ($noticeDismissButton.length)
+                {
+                    $noticeDismissButton.trigger('click');
+                }
+
+                this.$target.text();
```

Comment:
Is this debug code to be removed? Or an attempt to blank the target text?

Reply:
That debug code should be gone now.